tests/system/marathon_common_tests.py

```diff
@@ -840,8 +840,8 @@ def test_app_update_rollback():
     deployment_wait(service_id=app_id)
 
     # update to 1 instance is rollback to 2
-    tasks = client.get_tasks(app_id)
-    assert len(tasks) == 2, "The number of tasks is {} after rollback, but 2 was expected".format(len(tasks))
+    assert_that(lambda: client.get_tasks(app_id),
```

Comment:
🤔 interesting. How come when the deployment is finished, the tasks endpoint still returns outdated info? @meln1k

Reply:
Very good point, @alenkacz. If that the case, this is clearly a breach in Marathons API contract and we should fix the underlying issue, and not its symptom.

Reply:
Oh, I think this is because our [LaunchStats](https://github.com/mesosphere/marathon/blob/95f5781ee47e8d19e4c8eac9d23ea9e45c35d5f1/src/main/scala/mesosphere/marathon/core/launchqueue/LaunchStats.scala) are async now and are a little behind the deployments. We had a flaky integration test because of that.

Reply:
..ah, that's just for our launch queue. Anyways, there can be a race condition. @meln1k, do you have the logs of a failed run that show that all instances are up and running?